client/src/pages/HomePage.js

```diff
@@ -32,21 +35,36 @@ const HomePage = () => {
       "/api/mentions?platforms=reddit" +
       (searchTerm ? "&keyword=" + searchTerm : "") +
       "&page=" +
-      newPage;
+      (currentPage + 1) +
+      "&sort=" +
+      sort;
 
     const results = await axios(url, config);
     setHasMore(results.data.nextPage ? true : false);
     const newData = [...mentionDatas, results.data.mentions].flat();
     setMentionDatas(newData);
+    setCurrentPage((prev) => prev + 1);
+  };
+
+  const handleAlignment = (event, newAlignment) => {
+    getMentions(dispatch, searchTerm, user.platforms, 1, newAlignment)
+      .then((data) => {
+        setMentionDatas(data.mentions);
+        setHasMore(data.nextPage ? true : false);
+      })
+      .catch((err) => alert(err.message));
+    setSort(newAlignment);
+    setCurrentPage(1);
```

Comment:
There's a delay when switching (which is to be expected) so maybe it would make sense to add a loading component while fetching the data

Reply:
Good point, i'll work on that

Reply:
done